adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -11,35 +11,40 @@ module Middlewares
         # TracerMiddleware propagates context and instruments Faraday requests
         # by way of its middlware system
         class TracerMiddleware < ::Faraday::Middleware
+          HTTP_METHODS_SYMBOL_TO_STRING = {
+            connect: 'CONNECT',
+            delete: 'DELETE',
+            get: 'GET',
+            head: 'HEAD',
+            options: 'OPTIONS',
+            patch: 'PATCH',
+            post: 'POST',
+            put: 'PUT',
+            trace: 'TRACE'
+          }.freeze
+
           def call(env)
-            return app.call(env) if disable_span_reporting?(env)
+            http_method = HTTP_METHODS_SYMBOL_TO_STRING[env.method]
 
-            tracer.in_span(env.url.to_s,
-                           attributes: { 'component' => 'http',
-                                         'http.method' => env.method,
-                                         'http.url' => env.url.to_s },
-                           kind: :client) do |span|
-              propagate_context(span, env)
+            tracer.in_span(
+              "HTTP #{http_method}",
+              attributes: {
+                'component' => 'http',
+                'http.method' => http_method,
+                'http.url' => env.url.to_s
+              },
+              kind: :client
+            ) do |span|
+              propagator.inject(span.context, env.request_headers)
```

Comment:
Since the context prop PR was merged this can be changed to:

```suggestion
             OpenTelemetry.propagation.inject(env.request_headers)
```